lib/ruby_lsp/requests/semantic_highlighting.rb

```diff
@@ -110,14 +110,16 @@ def run
         @encoder.encode(@tokens)
       end
 
-      sig { override.params(node: SyntaxTree::Call).void }
+      sig { override.params(node: SyntaxTree::CallNode).void }
       def visit_call(node)
         return super unless visible?(node, @range)
 
         visit(node.receiver)
 
         message = node.message
-        add_token(message.location, :method) if message != :call
+        if message != :call && message&.value != "lambda"
```

Comment:
Why do we need to check for lambda here? Should we be checking for special methods instead?

Reply:
Without the `lambda` check, we get a failure on `test/expectations/semantic_highlighting/lambdas.exp`.

I've updated to use the `special_methods?` check.